doc/mgmt/SONiC-Management-Framework-Image-mgmt-HLD.md

```diff
@@ -0,0 +1,250 @@
+# Image Management
+Software upgrade and image installation support for SONiC
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev  |    Date    |      Author       | Change Description |
+| :--: | :--------: | :---------------: | ------------------ |
+| 0.1  | 09/13/2019 | Arunsundar Kannan | Initial version    |
+
+# About this Manual
+This document provides general information about the Image management and installation feature implementation in SONiC.
+# Scope
+Covers northbound interface for the Image Management feature, as well as unit test cases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**         | **Meaning**                    |
+| ---------------- | ------------------------------ |
+| Image Management | Image installation, validation |
+
+# 1 Feature Overview
+
+Provide management framework capabilities to handle:
+
+- Image Installation
+- List available images
+- Removal of an available image
+- Set default image for next boot
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide management framework support to existing SONiC capabilities with respect to Image Management
+
+### 1.1.2 Configuration and Management Requirements
```

Comment:
Provide a list of REST API based on the data models supported.

Reply:
Done